test/local_test_ckpt_index_file.py

```diff
@@ -0,0 +1,170 @@
+from typing import List
+import unittest
+import argparse
+import shutil
+import json
+import os
+
+from pippy.IR import pipe_split, TrivialLossWrapper
+from pippy.compile import compile_stage
+import torch.distributed as dist
+import torch.optim as optim
+import torch
+
+d_hid = 512
+chunk_size = 256
+
+
+class ExampleCode(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.mm_param0 = torch.nn.Parameter(torch.randn(d_hid, d_hid))
+        self.mm_param1 = torch.nn.Parameter(torch.randn(d_hid, d_hid))
+        self.mm_param2 = torch.nn.Parameter(torch.randn(d_hid, d_hid))
+        self.lin0 = torch.nn.Linear(d_hid, d_hid)
+        self.lin1 = torch.nn.Linear(d_hid, d_hid)
+
+    def forward(self, x):
+        x = torch.mm(x, self.mm_param0)
+        skip_connection = x
+        x = torch.relu(x)
+        pipe_split()
+        x = torch.mm(x, self.mm_param1)
+        x = self.lin0(x)
+        pipe_split()
+        x = torch.relu(x)
+        x = x + skip_connection
+        x = torch.mm(x, self.mm_param2)
+        pipe_split()
+        x = self.lin1(x)
+        x = torch.relu(x)
+        return x
+
+
+def run_worker(args: List[str | int]) -> None:
+    ec = ExampleCode()
+    loss_fn = torch.nn.MSELoss(reduction="sum")
+    ec_with_loss = TrivialLossWrapper(ec, loss_fn)
+    ec_with_loss.to(args.device)
+
+    ec_x = torch.randn(args.chunks * chunk_size, d_hid, device=args.device)
+    target = torch.randn(args.chunks * chunk_size, d_hid, device=args.device)
+
+    stage = compile_stage(
+        ec_with_loss,
+        args.rank,
+        args.world_size,
+        args.chunks,
+        args.device,
+        None,
+        [ec_x, target],
+    )
+
+    # Create an optimizer for stage submodule's parameters
+    optimizer = optim.SGD(stage.submod.parameters(), lr=1e-3, momentum=0.9)
+
+    for _ in range(2):
+        # Zero gradients
+        optimizer.zero_grad()
+
+        # Run
+        if args.rank == 0:
+            stage(ec_x)
+        elif args.rank == args.world_size - 1:
+            stage(target)
+        else:
+            stage()
+
+        # Take an optimization step
+        optimizer.step()
+
+    dist.barrier()
+    print(f"Rank {args.rank} completes")
+
+    # Last rank checks index file
+    if args.rank == args.world_size - 1:
```

Comment:
Could we add a comment that we only need one rank to check the index file? 
Likely, during the actual saving, we will be using rank0 to generate the index file. Let's use rank 0 to check this instead.